controllers/rack.go

```diff
@@ -646,6 +646,12 @@ func (r *SingleClusterReconciler) upgradeRack(statefulSet *appsv1.StatefulSet, r
 		}
 	}
 
+	err = r.updateSTSHeadlessSvc()
```

Comment:
Right now headless svc is created once in `createEmptyRack` func for the 1st rack and skipped for the rest of the racks. As this service is tied to one CR and not rack specific, can we move this existing create code and new update code out of rack processing  and call only once in every reconcile like `createorUpdateHeadlessService`.
This way we don't have to call create and update multiple times at different places